spec/workers/webhook_worker_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe WebhookWorker do
+  describe "perform" do
+    let(:report) { { some: "json" } }
+    let(:webhook_uri) { "http://webhooks-rule.org/webhook" }
+    let(:secret_key) { nil }
+
+    context "with a secret key" do
+      let(:secret_key) { "this is a secret key" }
+
+      # http://www.freeformatter.com/hmac-generator.html
+      let(:expected_signature) { "d44d805a16b90922edb48f14ff4e858a5d3e39bd" }
```

Comment:
Hmm, yes, I wasn't sure about this one. I went for the website approach so we can be 100% sure the HMAC is generated properly. I put the JSON in with the key and got the expected response.

If I just copy in the code from the signature generation method, then I would say we're not really testing that we get a valid HMAC situation, we're just running the same code twice.